pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -417,18 +419,62 @@ func (a *ApplicationBackupController) handle(ctx context.Context, backup *stork_
 	// to new map in case of csi based backups
 	if snapshotClassName, ok := backup.Spec.Options[optCSISnapshotClassName]; ok && len(backup.Spec.CSISnapshotClassMap) == 0 {
 		vsc, err := externalsnapshotter.Instance().GetSnapshotClass(snapshotClassName)
-		if err != nil {
+		// In Case of Default given, list out all csi drivers and their snapshot classes and make a map of default vsc to its respective csi driver
+		// make empty map if no default vsc found
+		if k8s_errors.IsNotFound(err) && (snapshotClassName == "default" || snapshotClassName == "Default") {
+			backup.Spec.CSISnapshotClassMap = make(map[string]string)
+
+			// list all csi drivers in the cluster
+			drivers, dErr := storage.Instance().ListCsiDrivers()
+			if dErr != nil {
+				log.ApplicationBackupLog(backup).Errorf("Error getting csi drivers: %v", dErr)
+				a.recorder.Event(backup,
+					v1.EventTypeWarning,
+					string(stork_api.ApplicationBackupStatusFailed),
+					dErr.Error())
+				return nil
+			}
+
+			// list all snapshot classes in the cluster
+			snapshotClasses, sErr := externalsnapshotter.Instance().ListSnapshotClasses()
+			if sErr != nil {
+				log.ApplicationBackupLog(backup).Errorf("Error getting volumesnapshotclasses: %v", sErr)
+				a.recorder.Event(backup,
+					v1.EventTypeWarning,
+					string(stork_api.ApplicationBackupStatusFailed),
+					sErr.Error())
+				return nil
+			}
+
+			for _, driver := range drivers.Items {
+				backup.Spec.CSISnapshotClassMap[driver.Name] = ""
+				driverCount := 0
+				for _, vsc := range snapshotClasses.Items {
+					if vsc.Driver == driver.Name {
+						driverCount++
+						if isTrue, exist := vsc.GetAnnotations()[defaultVolumeSnapshotClassAnnotation]; isTrue == "true" && exist {
+							backup.Spec.CSISnapshotClassMap[driver.Name] = vsc.Name
+						}
+					}
+				}
+				// if no default vsc found and there's only one vsc for that provisioner we will use that as default
+				if driverCount == 1 && backup.Spec.CSISnapshotClassMap[driver.Name] == "" {
+					backup.Spec.CSISnapshotClassMap[driver.Name] = snapshotClasses.Items[0].Name
```

Comment:
what will be the map in case of snpashotClasses.Items[0].Name != vsc.Driver
but snapshotClasses.Items[2].Driver == vsc.Driver.
WIll this line still be good ?

I think you should update line 454 to this

driverCount++ 
If driverCount == 1 {
     savedOnlydriverName = vsc.Name
}

And line 462 will become
backup.Spec.CSISnapshotClassMap[driver.Name] = savedOnlyDriverName

Reply:
if driverCount == 1 that means it belong to that driver ... on line 451 i am updating the driverCount to 0 for each driver so snpashotClasses.Items[0].Name != vsc.Driver this sceanario will never come

Reply:
if there's no default and we have 2 or more vsc we will not update the map

Reply:
This code looks buggy to me. SnapshotClasses is not directly mapped to drivers.Items. I am unable to understand how driverCount++ is always have 1 : 1 mapping to snapshotClasses.Items[0]